[IM] Refactor component templates API integration tests

alisonelizabeth opened the pull request:
Fixes https://github.com/elastic/kibana/issues/67969

This PR refactors the component_templates API integration tests. It now handles any promise rejections when creating/deleting templates in the setup/teardown of tests and logs the error.

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51836/)
* Commit: a0aa645b13be829116b75dc525a08a28921a05db

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51836/","number":"51836","commit":"a0aa645b13be829116b75dc525a08a28921a05db"}]}
PIPELINE-->

elasticmachine commented:
Pinging @elastic/es-ui (Team:Elasticsearch UI)

alisonelizabeth commented:
@elasticmachine merge upstream

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52293/)
* Commit: e161bd2f65bb566c31aab7bd4700e3e690942028

### History
* :green_heart: [Build #51836](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51836/) succeeded a0aa645b13be829116b75dc525a08a28921a05db

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51836/","number":"51836","commit":"a0aa645b13be829116b75dc525a08a28921a05db"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52293/","number":"52293","commit":"e161bd2f65bb566c31aab7bd4700e3e690942028"}]}
PIPELINE-->

sebelga commented on x-pack/test/api_integration/apis/management/index_management/component_templates.ts:
```diff
@@ -240,29 +267,40 @@ export default function ({ getService }: FtrProviderContext) {
+
+      before(() => {
+        // Create several component templates that can be used to test deletion
+        Promise.all(
```
I am surprised that this works without returning the Promise 😊

sebelga commented on x-pack/test/api_integration/apis/management/index_management/lib/elasticsearch.js:
```diff
@@ -34,14 +35,29 @@ export const initElasticsearchHelpers = (es) => {
   };
 
+  const cleanUpComponentTemplates = () => {
+    Promise.all(componentTemplatesCreated.map(deleteComponentTemplate))
```
It would be better to return the Promise.

alisonelizabeth commented:
Thanks for the review @sebelga! I believe I addressed all of your concerns. Would you mind taking another look?

> I left a few comments about not throwing errors when the resource was not created 

👍 Definitely makes sense. Fixed.

> and Promises not returned.

🤦‍♀️ Thanks for catching this! Fixed as well.

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52792/)
* Commit: b12aa242df6dd89ddb586fbe952259e758d80a18

### History
* :green_heart: [Build #52293](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52293/) succeeded e161bd2f65bb566c31aab7bd4700e3e690942028
* :green_heart: [Build #51836](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51836/) succeeded a0aa645b13be829116b75dc525a08a28921a05db

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/51836/","number":"51836","commit":"a0aa645b13be829116b75dc525a08a28921a05db"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52293/","number":"52293","commit":"e161bd2f65bb566c31aab7bd4700e3e690942028"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/52792/","number":"52792","commit":"b12aa242df6dd89ddb586fbe952259e758d80a18"}]}
PIPELINE-->
